### PR TITLE
SPH-684 - Mits overrides (evt obv geometry)

### PR DESCRIPTION
**Pull request by JRdeLange**
- `tool_by_tool_walkthrough.ipynb`:
  - Ik heb een override toegevoegd aan een van de demo-cellen
- `test_override_criteria.py`
  - Tests voor het OverrideCriterium - hij heeft genoeg tests voor zn eigen test file vind ik. Als je nog een mooie edge case weet laat maar weten.
- `test_tool_by_tool_walkthrough.ipynb`
  - Test toegevoegd waarin het effect van een geometry OverrideCriterium getest wordt
- `criteria.py`
  - OverrideCritieium toegevoegd. Bij het aanmaken met een GeoSeries als override_geometry wordt deze geserialiseerd. Verder functioneert bijna het net als de andere broncriteria - alleen ipv FGRtypen/bodemcodes krijgt het de geometry van de shape waar het bij hoort mee. Hiermee kunnen geometry OverrideCriteria gecheckt worden.
- `definitietabel.py`
  - Heeft nu een override_dict met daar in mitsstring -> OverrideCriterium. Deze wordt geraadpleegd bij het aanmaken van de habitatvoorstellen. Zit de mitstekst in de dict? Dan worden niet de criteria uit de definitietabel gepakt, maar het bijbehorende OverrideCriterium.
- `common.py`
  - override_dict in de params voor de tool_by_tool tests
- `main.py`
  - override_dict in de deftabel gooien
- `vegkartering.py`
  - Comment voor de duidelijkheid

### CODE REVIEW

**Comment by mark-boer** [comment]
Ik zal er naar kijken als de serialisatie issues gefixd zijn.

File: veg2hab/main.py [hunk 1]
```diff
@@ -171,6 +171,9 @@ def run_3_definitietabel_en_mitsen(params: ApplyDefTabelInputs):
     logging.info("Kartering is succesvol ingelezen")
 
     deftabel = DefinitieTabel.from_excel(Path(constants.DEFTABEL_PATH))
+    # @Mark
+    # Hier evt vertaling van hoe je override dict doorgeeft naar een Dict[str, OverrideCriterium]
+    deftabel.set_override_dict(params.override_dict)
```
[the remark this answers is not in the document]
Comment (mark-boer) [reply veg2hab/main.py:176]: Ik denk dat ik er wel uit kom. Moeten alleen even gaan bedenken hoe we dit het handigst kunnen doen in

**Review by mark-boer — APPROVED** [review]

File: tests/test_override_criteria.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+    assert crit1.cached_evaluation == MaybeBoolean.TRUE
+    assert crit2.cached_evaluation == MaybeBoolean.TRUE
+    assert crit3.cached_evaluation == MaybeBoolean.FALSE
+
+
+def test_with_multipolygon(split_override_geometry, kartering_geometry):
+    """
+    +-------+-------+-------+
+    | 0     | 1     | 2     |
```
[the remark this answers is not in the document]
Comment (JRdeLange) [reply tests/test_override_criteria.py:185]: ik vind ze veel te leuk om te maken

File: tests/test_override_criteria.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+@pytest.fixture(scope="module")
+def definitietabel():
+    return DefinitieTabel.from_excel(Path(constants.DEFTABEL_PATH))
+
+
+def test_override_criteria_instantiation(override_geometry):
+    # non-geometry OverrideCriterium
+    crit = OverrideCriterium(mits="mitsemits", truth_value=MaybeBoolean.TRUE)
+    assert isinstance(crit, OverrideCriterium)
```
[the remark this answers is not in the document]
Comment (JRdeLange) [reply tests/test_override_criteria.py:55]: ?
